test/extensions/filters/http/ext_proc/filter_test.cc

```diff
@@ -1068,8 +1068,6 @@ TEST_F(HttpFilterTest, PostAndChangeRequestBodyBufferedComesFast) {
   Buffer::OwnedImpl buffered_data;
   setUpDecodingBuffering(buffered_data);
 
-  // Buffering and callback isn't complete so we should watermark
-  EXPECT_CALL(decoder_callbacks_, onDecoderFilterAboveWriteBufferHighWatermark());
```

Comment:
All the test in this PR are removing the wrong (I think) mock expectation. 

watermarks should not be invoked for small data buffering.